app/grandchallenge/reader_studies/serializers.py

```diff
@@ -115,13 +115,20 @@ class DisplaySetSerializer(HyperlinkedModelSerializer):
 
     def get_index(self, obj) -> int | None:
         if obj.reader_study.shuffle_hanging_list:
-            try:
-                return self.context["view"].randomized_qs.index(obj)
-            except ValueError:
-                # The list is empty if no reader study is specified.
-                return None
+            return obj.shuffled_index
         else:
-            return obj.standard_index
+            try:
+                row_number = obj.row_number
+            except AttributeError:
+                # The annotation wasn't made when getting this object
+                row_number = (
+                    DisplaySet.objects.with_row_number()
+                    .get(pk=obj.pk)
+                    .row_number
+                    - 1
+                )
+
+            return row_number - 1
```

Comment:
Here you subtract 1 twice for cases where the annotation wasn't made when getting the object, right?

Reply:
Ah I guess that's because the RowNumber() window function returns 1-based numbering, not 0-based.